ethcore/types/src/receipt.rs

```diff
@@ -60,10 +64,32 @@ impl Receipt {
             outcome,
         }
     }
-}
+    pub fn decode(rlp: &Rlp) -> Result<Self, DecoderError> {
+        if rlp.item_count()? == 3 {
```

Comment:
`if rlp.item_count()? == 0` could crash?

Reply:
for length `1` it will fail in `else` branch and that is okay. But for `rlp.item_count()? == 5` or if it greater than 5 it will pass but still be invalid. I think it is okay for us to be strict here and check the exact lenght.